[Jannice Suciptono] iP

jannnice opened the pull request:
# POPOOH
Yes, the file name is Duke. I forgot to change it at the start, oops :)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
test 1

Commit:
Level 0. Rename, Greet, Exit

Commit:
Level 1. Echo

Commit:
Level 2. Add, List

Commit:
Level 3. Mark as Done

Commit:
Level 4. ToDos, Events, Deadlines

Commit:
Level 5. Handle Errors

Commit:
Level 6. Delete

Commit:
Added enum

Commit:
Level 7. Save

Commit:
Level 8. Dates and Times

Commit:
A-MoreOOP

Commit:
A-Packages

Commit:
Merge branch 'add-gradle-support'

Commit:
A-Gradle

Commit:
A-JUnit

Commit:
A-JavaDoc

Commit:
Level 9. Find

Commit:
A-CodingStandard

Commit:
Merge branch 'branch-Level-9'

Commit:
Merge branch 'branch-A-CodingStandard'

GSgiansen commented on src/main/java/Exceptions/DukeException.java:
```diff
@@ -0,0 +1,9 @@
+package  Exceptions;
```
Minor issue but extra white space!

GSgiansen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,122 @@
+    private static TaskList tasks;
+    private static String directoryPath;
+    private static String filePath;
+    protected static final String horizontalLine = "    ____________________________________________________________";
```
Constant names should be in all Caps!

GSgiansen commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,122 @@
+                ui.exit();
+                break;
+            case UNMARK:
+                System.out.println(horizontalLine);
```
I think the print statements would be better in the Ui class rather than printing in the Duke class

GSgiansen commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,105 @@
+    }
+    // saves new tasks
+    protected void saveTask(ArrayList<Task> taskList) throws IOException {
+        FileWriter fw = new FileWriter(this.filePath);
```
Perhaps a better variable name be used for code readability ?

GSgiansen commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,55 @@
+    public void markAsDone() {
+        this.isDone = true;
+        System.out.println("     Nice! I've marked this task as done:\n"
+                           + "       " + this.printDesc());
```
I think this is more than the 8 spaces recommended!

GSgiansen commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+        numTask--;
+        System.out.println("     Noted. I've removed this task:\n"
+                + "     " + removed.printDesc() + "\n"
+                + "     Now you have " + numTask +" tasks in the list.");
```
Spacing should be 8 pages!

GSgiansen commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,124 @@
+        }
+        System.out.println("     Here are the matching tasks in your list:\n");
+        for (int a = 0; a < keywordTasks.size(); a++) {
+            System.out.println("     " + (a + 1) + ". " + keywordTasks.get(a).printDesc());
```
I would think for better abstraction it would be better to place the print statements in the Ui class

GSgiansen commented on src/test/java/duke/ParseTest.java:
```diff
@@ -0,0 +1,15 @@
+    public void deadlineDetails_success() {
+        Parser parser = new Parser();
+        String deadline = "homework /by: 2023-12-01T12:20:00";
+        String[] output = parser.deadlineDetails(deadline);
```
To get the deadline details, renaming the method to getDeadlineDetails would fit the naming conventions better

GSgiansen reviewed (commented):
Overall I think your implementation is great! Some changes to coding standard to enhance readability would help make your code more readable to others 🥳

Commit:
A-CheckStyle

Commit:
Setting up the JavaFX

Commit:
Setting up of GUI

Commit:
Created Jar File

Commit:
Addition of Java Assert feature

Java assert feature is not added.

Java assert feature allows us to detect potential bugs as well as
locating the position of the bug.

Let's add Java Assert feature to monitor our codes.

Commit:
A-CodeQuality

Some string messages are repeated and long.

The code became very long, making it seems like a complicated
code when it is not.

Let's rewrite the repeating messages as constants to make the
code neater.

These changes will make the code more presentable and reader
friendly.

jannnice committed:
Merge pull request #2 from jannnice/branch-A-CodeQuality

Branch a code quality

jannnice committed:
Merge pull request #1 from jannnice/branch-A-Assertions

Merging with branch A-Assertions

Commit:
BCD-Extension

Commit:
A-BetterGui

Commit:
A-Personality

Commit:
A-MoreErrorHandling

Added in some more error handling when there is missing arguments.

Commit:
Improvements to the GUI

Commit:
A-UserGuide

Commit:
Addition of some Javadoc comments to the code
